src/parrot_mcp_server/security_core.py

```
@@ -0,0 +1,431 @@
+    # ------------------------------------------------------------------
+
+    @staticmethod
+    def _build_profile(environ: dict[str, Any]) -> RequestProfile:
+        path = environ.get("PATH_INFO", "/")
+        header_size = sum(
+            len(k) + len(str(v)) + 4
+            for k, v in environ.items()
+            if k.startswith("HTTP_")
+        )
+        return RequestProfile(
+            source_ip=environ.get("REMOTE_ADDR", "0.0.0.0"),
```

Ruff (0.15.5)

[error] Possible binding to all interfaces

(S104)